### ISSUE TITLE
ActionExtension for events does not respect guard proeprty

### ISSUE THREAD

**Issue by Tyrrrz** [canton7/Stylet#55]
From wiki: 
>The syntax is exactly the same, although there's no concept of a guard property here.

Why? I want to be able to set `MouseLeftButtonUp="{s:Action SomeAction}"` and only have the handler actually execute when the guard property lets it. The justification is that I want to be able to use events in the same way I use commands, without having to use some kind of `EventToCommand` contraption.

**Comment by canton7** [comment]
Because events have no concept of "can execute". ICommand has the CanExecute method, but events have nothing of the sort.

If you want to prevent your event handler from running in particular cases, put the code `if (!CanSomeAction) return` at the top.

**Comment by Tyrrrz** [comment]
That's true that the control cannot (and should not) be disabled. That said, from view-model level of abstraction, since Stylet effectively removes the concept of `ICommand` (which is amazing), the `CanXYZ` guard properties should no longer be tied to `ICommand.CanExecute`. As a person using Stylet, I greatly enjoy the fact that I don't have to bother with `ICommand` or its implementation and as such I prefer not to think about it. To me, `CanXYZ` is a property that simply answers "can XYZ be executed?" without having anything to do with controls being disabled or not. I would say controls being disabled is more like an appreciated side effect.
